brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -68,6 +68,13 @@ static Builder builder(LocalTracer source) {
     }
 
     /**
+     * Clears current span.
+     */
+    public void clearCurrentSpan() {
```

Comment:
This method clearCurrentSpan() only exists on ServerTracer: it is
because of setStateCurrentTrace(), used when a span is received from
headers.

That use case doesn't exist for ClientTracer or LocalTracer.

On Fri, Aug 12, 2016 at 2:22 PM, mck <notifications@github.com> wrote:
> In brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java:
>
>> @@ -68,6 +68,13 @@ static Builder builder(LocalTracer source) {
>>      }
>>
>>      /**
>> +     * Clears current span.
>> +     */
>> +    public void clearCurrentSpan() {
>
> i'm not sure why i'd want to use localSpanThreadBinder instead of
> localTracer here?
>
> Given that the other tracers (serverTracer and clientTracer) have this
> method doesn't it make sense that localTracer also for the sake of
> consistency has it?
>
> Then the effort to reduce API (/cognitive load) can be done separately, but
> equally to all three tracers?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.